frontend/src/static/js/components/webstatus-form-date-range-picker.ts

```diff
@@ -0,0 +1,233 @@
+/**
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  SlButton,
+  SlChangeEvent,
+  SlInput,
+  SlInputEvent,
+} from '@shoelace-style/shoelace';
+import {CSSResultGroup, LitElement, PropertyValues, css, html} from 'lit';
+import {customElement, property, query, state} from 'lit/decorators.js';
+import {SHARED_STYLES} from '../css/shared-css.js';
+
+export interface DateRangeChangeEvent {
+  startDate: Date;
+  endDate: Date;
+}
+
+/**
+ * @summary Date range picker
+ * @event CustomEvent<DateRangeChangeEvent> webstatus-date-range-change - Emitted when the the date range is changed.
+ * @property {Date} minimumDate - The minimum selectable date. **Required.**
+ * @property {Date} maximumDate - The maximum selectable date. **Required.**
+ * @property {Date} startDate - The initial start date for the range. **Required.**
+ * @property {Date} endDate - The initial end date for the range. **Required.**
+ */
+@customElement('webstatus-form-date-range-picker')
+export class WebstatusFormDateRangePicker extends LitElement {
+  @property({type: Object})
+  minimumDate?: Date;
+
+  @property({type: Object})
+  maximumDate?: Date;
+
+  @property({type: Object})
+  startDate?: Date;
+
+  @property({type: Object})
+  endDate?: Date;
+
+  @query('#start-date')
+  readonly startDateEl?: SlInput;
+
+  @query('#end-date')
+  readonly endDateEl?: SlInput;
+
+  @query('#date-range-picker-btn')
+  readonly submitBtn?: SlButton;
+
+  @state()
+  private _pendingStartDate = false;
```

Comment:
"Pending" sounds like you are waiting for the value to be entered.  You could add a comment to clarify that this is actually pending being finalized by the user pressing submit.  Or, it might be more clear as something like `_startIsDirty` or `_startHasChanged`.  I see you added a comment about it to `isSubmitButtonEnabled()`, but is pretty far down in the file.

Reply:
Good point! I will change them to _startHasChanged and _endHasChanged